components/BottomNavigation/tests/snapshot/MDCBottomNavigationBarSnapshotTests.m

```diff
@@ -1093,4 +1093,90 @@ - (void)testBarItemsBottomAnchorWithoutOffset {
   }
 }
 
+#pragma mark - KVO tests
+
+- (void)testChangeSelectedIconWhenUnselected {
+  // Given
+  [self configureBottomNavigation:self.navigationBar
+                    withAlignment:MDCBottomNavigationBarAlignmentJustified
+                  titleVisibility:MDCBottomNavigationBarTitleVisibilityAlways
+                  traitCollection:nil
+                        allTitles:kLongTitleLatin];
+  self.navigationBar.frame = CGRectMake(0, 0, kWidthiPad, kHeightTypical);
```

Comment:
_Tip:_
```
- (void)setupNavigationBarForSelectedSnapshotTest {
  self.navigationBar.frame = CGRectMake(0, 0, kWidthiPad, kHeightTypical);
  [self performInkTouchOnBar:self.navigationBar item:self.tabItem1];
  self.navigationBar.selectedItemTintColor = UIColor.orangeColor;
  self.navigationBar.unselectedItemTintColor = UIColor.blackColor;
}
```

Then call that in all four test. It would be much easier for me to read if I could distinctly follow the differences between all 4 tests.

Reply:
That's why I put them in the "Given" section. I prefer to keep tests as explicit and self-contained as practical (avoiding boilerplate). I'll file a backlog issue to clean up the snapshot tests file in general, though and I can work on this as one of the items.  It may be more reasonable to split the tests into separate files.

Reply:
Opened https://github.com/material-components/material-components-ios/issues/6844 to track the work - I'll try to get it done this week.